pkg/doc/verifiable/example_credential_test.go

```diff
@@ -182,3 +180,112 @@ func ExampleCredential_extraFields() {
 	// {"@context":["https://www.w3.org/2018/credentials/v1","https://www.w3.org/2018/credentials/examples/v1"],"credentialSchema":[],"credentialSubject":{"degree":{"type":"BachelorDegree","university":"MIT"},"id":"did:example:ebfeb1f712ebc6f1c276e12ec21","name":"Jayden Doe","spouse":"did:example:c276e12ec21ebfeb1f712ebc6f1"},"expirationDate":"2020-01-01T19:23:24Z","id":"http://example.edu/credentials/1872","issuanceDate":"2010-01-01T19:23:24Z","issuer":{"id":"did:example:76e12ec712ebc6f1c221ebfeb1f","name":"Example University"},"referenceNumber":83294847,"type":["VerifiableCredential","UniversityDegreeCredential"]}
 }
+
+//nolint:lll
+func ExampleNewCredential() {
+	// Issuer is about to issue the university degree credential for the Holder
+	vcEncoded := &verifiable.Credential{
+		Context: []string{"https://www.w3.org/2018/credentials/v1", "https://www.w3.org/2018/credentials/examples/v1"},
+		ID:      "http://example.edu/credentials/1872",
+		Types:   []string{"VerifiableCredential", "UniversityDegreeCredential"},
+		Subject: UniversityDegreeSubject{
+			ID:     "did:example:ebfeb1f712ebc6f1c276e12ec21",
+			Name:   "Jayden Doe",
+			Spouse: "did:example:c276e12ec21ebfeb1f712ebc6f1",
+			Degree: UniversityDegree{
+				Type:       "BachelorDegree",
+				University: "MIT",
+			},
+		},
+		Issuer: verifiable.Issuer{
+			ID:   "did:example:76e12ec712ebc6f1c221ebfeb1f",
+			Name: "Example University",
+		},
+		Issued:  &issued,
+		Expired: &expired,
+		Schemas: []verifiable.TypedID{},
+		CustomFields: map[string]interface{}{
+			"referenceNumber": 83294847,
+		},
+	}
+
+	// ... in JWS form.
+	jwtClaims, err := vcEncoded.JWTClaims(true)
+	if err != nil {
+		fmt.Println(fmt.Errorf("failed to marshal JWT claims of VC: %w", err))
+	}
+
+	jws, err := jwtClaims.MarshalJWS(verifiable.EdDSA, privIssuerKey, "")
+	if err != nil {
+		fmt.Println(fmt.Errorf("failed to sign VC inside JWT: %w", err))
+	}
+
+	// The Holder receives JWS and decodes it.
+	_, vcDecodedBytes, err := verifiable.NewCredential(
+		[]byte(jws),
+		verifiable.WithPublicKeyFetcher(verifiable.SingleKey(privIssuerKey.Public())))
+	if err != nil {
+		fmt.Println(fmt.Errorf("failed to decode VC JWS: %w", err))
+	}
+
+	fmt.Println(string(vcDecodedBytes))
+	// The Holder then e.g. can save the credential to her personal verifiable credential wallet.
+
+	// Output: {"@context":["https://www.w3.org/2018/credentials/v1","https://www.w3.org/2018/credentials/examples/v1"],"credentialSchema":[],"credentialSubject":{"degree":{"type":"BachelorDegree","university":"MIT"},"id":"did:example:ebfeb1f712ebc6f1c276e12ec21","name":"Jayden Doe","spouse":"did:example:c276e12ec21ebfeb1f712ebc6f1"},"expirationDate":"2020-01-01T19:23:24Z","id":"http://example.edu/credentials/1872","issuanceDate":"2010-01-01T19:23:24Z","issuer":{"id":"did:example:76e12ec712ebc6f1c221ebfeb1f","name":"Example University"},"referenceNumber":83294847,"type":["VerifiableCredential","UniversityDegreeCredential"]}
```

Comment:
can this be pretty printed on multiple lines? I guess by doing so, you won't need `\\nolint:lll` at the top of the function

Reply:
I don't think so, as when running the example it makes the exact matching of std output with `// Output` content.